### PR TITLE
Add optional exception attribute to ErrorArtifact

### PR DESCRIPTION
**Pull request by michal-repo**
As discussed on discord channel back in March (handling content management policy error), adding optional exception attribute to ErrorArtifact.
This will allow to check Exception instance against eg. OpenAI exceptions to handle it properly in your application.

### CODE REVIEW

**Review by collindutter — COMMENTED** [review]
Fantastic, thank you for the PR! Could you also please add [here](https://github.com/griptape-ai/griptape/blob/f0a7c9449d8c00efa87d2b4807855d77ab7add11/griptape/tasks/actions_subtask.py#L108)?

**Comment by michal-repo** [comment]
Sure thing, added.

**Comment by collindutter** [comment]
Ah could you also add a section in `CHANGELOG.md` `Added` section?

Something like: 
```
- Optional `exception` field to `ErrorArtifact`.
```

**Review by collindutter — APPROVED** [review]
Looks like you just need to merge the latest base branch changes into your branch. Then we're good!

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `output_task_input` and `output_task_output` fields to `FinishStructureRunEvent`.
 - `AmazonS3FileManagerDriver` for managing files on Amazon S3.
 - `MediaArtifact` as a base class for `ImageArtifact` and future media Artifacts.
+- Added optional `exception` field to `ErrorArtifact`.
```
[the remark this answers is not in the document]
Comment (michal-repo) [reply CHANGELOG.md:26]: ok

**Review by collindutter — APPROVED** [review]

**Pull request merged**